### ISSUE TITLE
Automatically enable high contrast theme when enabled in Windows

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#15360]
Newer electron version gives us API to find out if Windows runs with HC theme and also adds an event to test the change.

Some early experiments:

![untitled](https://cloud.githubusercontent.com/assets/900690/20218752/64cd0dd2-a826-11e6-99a0-ada545ace589.gif)

The only question is, do we ever expect a user to have Windows HC theme enabled but not in Code. And also the other way around.

Maybe @stevencl @bgashler1 @seanmcbreen @chrisdias have some experiences from how it works in VS.

**Comment by bpasero** [comment]
**integration.ts**
```typescript
// High Contrast Events
let nonHCTheme = this.themeService.getColorTheme();
ipc.on('vscode:enterHighContrast', (event) => {
	this.partService.joinCreation().then(() => {
		const currentColorTheme = this.themeService.getColorTheme();
		if (currentColorTheme !== 'hc-black') {
			nonHCTheme = this.themeService.getColorTheme();
		}
		this.themeService.setColorTheme('hc-black', false);
	});
});

ipc.on('vscode:leaveHighContrast', (event) => {
	this.partService.joinCreation().then(() => {
		this.themeService.setColorTheme(nonHCTheme, false);
	});
});
```

**windows.ts**
```typescript
// React to HC color scheme changes (Windows)
if (platform.isWindows) {
	systemPreferences.on('inverted-color-scheme-changed', () => {
		if (systemPreferences.isInvertedColorScheme()) {
			this.sendToAll('vscode:enterHighContrast');
		} else {
			this.sendToAll('vscode:leaveHighContrast');
		}
	});
}
```

**Comment by stevencl** [comment]
It would seem that VS switches to high contrast theme as soon as Windows changes and then doesn't allow you to change theme to anything else. When Windows is not in high contrast theme, it appears that you can't set VS in high contrast and can only choose the blue, light or dark themes.

**Comment by bpasero** [comment]
My idea was to:
* switch to HC theme if windows switches to HC theme while the app is running
* switch back to the previously used theme when windows switches back
* enable the HC theme if you start VS Code in windows HC mode

I would still allow you to pick the HC theme without running windows in HC mode. And I would also allow you to change the theme anytime away from HC even if windows is running in HC mode.

The only thing that might annoy people is the fact that we startup in HC theme when windows runs in HC mode even if the user previously maybe disabled HC theme in VS Code. Not sure if that would be an issue.

**Comment by bpasero** [comment]
Verification:
* starting Code when Windows is turned to HC mode opens Code in HC mode
* changing Windows to HC mode while Code is running switches Code into HC mode
